### ISSUE TITLE
Charts 11.7 - Customization

### ISSUE THREAD

**Issue by todayifeeltired** [HanSolo/charts#74]
Hello :)
I rellay like the CoxCombChart, but i see no way of customizing it (or am i wrong?!).

Basically i´d like to change the MouseEvent from MOUSE_PRESSED to MOUSE_ENTERED.
I´d also like to add something to the "...%" text on the ChartItems and change the size and text 
on the canvas showing after the MOUSE_PRESSED-Event.

**Comment by HanSolo** [comment]
Ok just to get it right, you would like to define your own mouse event handler? So I could add something like onMousePressed(), onMouseReleased(), onMouseEntered() and onMouseExited() to the chart. For the text on the chartitems I could add a method to define a formatString e.g. "%.0f%%" for percentage or whatever you like but it will only format the value. Would that help?

**Comment by HanSolo** [comment]
I've implemented parts of your request with commit 05c00527c775af3b2fff22bd1f38b8fd92dd8000 in the jdk17 branch. Mouse entered is not that easy because the Canvas is just one node. Instead I've added MouseMoved and a method that returns the selected item for a given MouseEvent. Please check the CoxcombChartTest in the jdk17 branch, there I've added the new features with an example.

**Comment by todayifeeltired** [comment]
Hello, thanks for the quick response and already done work.
I´ll try to have a look on it today.

But yeah, to clarify
I´d like to have the canvas-showing event as an on Mouse_Entered-Event, but Mouse_Moved is just fine.
On the ChartItem i´d like to also display the name of the item.
And on the canvas i´d like to show 3 rows of information and the opportunity to name them "as i want" and display the information "i want". And maybe to resize it a bit (or to set a custom canvas).

I dont know whether als this is possible /just too much work and i was wondering if you just tested in on the jdk17 branch or if you´d copy it on the jdk11 branch if everything is set and done.
Thank you very much for the work you put in!

**Comment by HanSolo** [comment]
Hi there, so I've added more stuff related to your request. 
- You now can show the item name if you like by using showItemName(true/false).
- You can decide whether you would like to use the ChartItem.textFill which enables you to use different text fills for each item by using setUseChartItemTextFill(true/false)

I've started to add the text thing you mentioned to the canvas but removed it again because it does not make sense. it is too specific. I suggest you add another layout container e.g. a VBox with 3 Labels, make it mouseTransparent(true) and add it on top of the chart using a StackPane. I've added this also to the CoxcombChartTest in the jdk17 branch so you can see how that works. This gives you the ability to overlay whatever you want without making the CoxcombChart getting features that are useful to you but maybe not for everyone :)
I could of course add the stuff also to the jdk11 branch once it works for you in the jdk17 branch.

**Comment by todayifeeltired** [comment]
Hi there :)
You are right, maybe this feature is not useful for everyone.

I´ve looked at your commits and i love it, its exactly what i want/need!

Would appreciate it, if you release it on jdk11.

Btw, for all the work i was causing: is there any way to paypal a beer/coffee to you?

**Comment by HanSolo** [comment]
Ok so I've backported the code to the main branch and released v11.11, might take some time to show up on Maven central. And there is also 17.1.7 which contains the same changes in the jdk17 branch. Both jars are also available via the github releases. Glad it works for you :)

**Comment by todayifeeltired** [comment]
Hi there (again) :D
Thanks again for the quick implementation of the issue. I am very happy with it :)

I have one more wish, if that's not too much to ask for.
I´d like to have another StringProperty on the chartitem (and another constructor).
Heres why:
i got some data that i want to show on the coxcombchart that has a very long name. 
i´d like to show a "succintly name" on the chart and on a label (where obviously would be enough space of showing it), the "long name".

Basically all i´d need is a stringproperty on the chartitem,  a constructor so i can fill it and a getter for the new StringProperty.

I can open another issue, but this seems more like an feature request (for which I did not find the possibility).
